openmetadata-ui/src/main/resources/ui/src/components/common/DisplayName/DisplayName.test.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ *  Copyright 2024 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import { act, fireEvent, render, screen } from '@testing-library/react';
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import DisplayName, { DisplayNameProps } from './DisplayName';
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  Link: jest
+    .fn()
+    .mockImplementation(({ children, ...props }) => (
+      <span {...props}>{children}</span>
+    )),
+}));
+
+jest.mock('../../../constants/constants', () => ({
+  DE_ACTIVE_COLOR: '#BFBFBF',
+  ICON_DIMENSION: { width: 16, height: 16 },
+}));
+
+jest.mock('../../Modals/EntityNameModal/EntityNameModal.component', () => {
+  return {
+    __esModule: true,
+    default: jest.fn(() => (
+      <div data-testid="entity-name-modal">Mocked Modal</div>
+    )),
+  };
+});
+
+const mockOnEditDisplayName = jest.fn();
+
+const mockProps: DisplayNameProps = {
+  id: '1',
+  name: 'Sample Entity',
+  displayName: 'Sample Display Name',
+  link: '/entity/1',
+  allowRename: true,
+  onEditDisplayName: mockOnEditDisplayName,
+};
+
+describe('Test DisplayName Component', () => {
+  it('Should render the component with the display name', async () => {
+    await act(async () => {
+      render(
+        <MemoryRouter>
+          <DisplayName {...mockProps} />
+        </MemoryRouter>
+      );
+
+      const displayNameField = await screen.getByTestId('column-display-name');
+
+      expect(displayNameField).toBeInTheDocument();
+      expect(displayNameField).toHaveTextContent('Sample Display Name');
+
+      const editButton = await screen.queryByTestId('edit-displayName-button');
+
+      expect(editButton).toBeInTheDocument();
+    });
+  });
+
+  it('Should render the component with name when display name is empty', async () => {
+    await act(async () => {
+      render(
+        <MemoryRouter>
+          <DisplayName {...mockProps} displayName={undefined} />
+        </MemoryRouter>
+      );
+
+      const nameField = await screen.getByTestId('column-name');
```

Comment:
We don't need `await` for `getByTestId`.

Reply:
Removed `await`.